app/pages/SearchResultsPage/SearchResultsPage.tsx

```diff
@@ -1,54 +1,108 @@
 import { useHistory, useLocation } from 'react-router-dom';
 import algoliasearch from 'algoliasearch/lite';
-import { InstantSearch, Configure, SearchBox } from 'react-instantsearch/dom';
-import qs from 'qs';
+import {
+  InstantSearch, Configure,
+} from 'react-instantsearch/dom';
+import qs, { ParsedQs } from 'qs';
+
+import { GeoCoordinates, useAppContext } from 'utils';
+
+import SearchResults from 'components/search/SearchResults/SearchResults';
+import Sidebar from 'components/search/Sidebar/Sidebar';
 
-import { useAppContext } from 'utils';
-import SearchResultsContainer from '../../components/search/SearchResultsContainer';
 import config from '../../config';
+import styles from './SearchResults.module.scss';
 
 const searchClient = algoliasearch(
   config.ALGOLIA_APPLICATION_ID,
   config.ALGOLIA_READ_ONLY_API_KEY,
 );
 
+/** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const SearchResultsPage = () => {
-  const history = useHistory();
-  const { search } = useLocation();
   const { userLocation } = useAppContext();
   const [lastPush, setLastPush] = useState(Date.now());
+  const { search } = useLocation();
+  const [expandList, setExpandList] = useState(false);
   const searchState = useMemo(() => qs.parse(search.slice(1)), [search]);
 
   return (
-    <div className="search-page-container">
-      <InstantSearch
-        searchClient={searchClient}
-        indexName={`${config.ALGOLIA_INDEX_PREFIX}_services_search`}
-        searchState={searchState}
-        onSearchStateChange={(nextSearchState: any) => {
-          const THRESHOLD = 700;
-          const newPush = Date.now();
-          setLastPush(newPush);
-          const newUrl = nextSearchState ? `search?${qs.stringify(nextSearchState)}` : '';
-          if (lastPush && newPush - lastPush <= THRESHOLD) {
-            history.replace(newUrl);
-          } else {
-            history.push(newUrl);
-          }
-        }}
-        createURL={(state: any) => `search?${qs.stringify(state)}`}
-      >
-        {userLocation ? (
-          <Configure aroundLatLng={`${userLocation.lat}, ${userLocation.lng}`} />
-        ) : (
-          <Configure aroundLatLngViaIP aroundRadius="all" />
-        )}
-        <div className="search-box">
-          <SearchBox />
-        </div>
-        <SearchResultsContainer />
-      </InstantSearch>
-    </div>
+    <InnerSearchReslts
+      history={useHistory()}
+      userLocation={userLocation}
+      lastPush={lastPush}
+      setLastPush={setLastPush}
+      expandList={expandList}
+      setExpandList={setExpandList}
+      searchState={searchState}
+    />
   );
 };
+
+/** Stateless inner component that just handles presentation. */
+const InnerSearchReslts = ({
+  history, userLocation, lastPush, setLastPush, expandList, setExpandList, searchState,
+}: {
+  history: any;
+  userLocation: GeoCoordinates;
+  lastPush: number;
+  setLastPush: Function;
```

Comment:
Let's specify the full function type signature for function types. `Function` doesn't tell us much, since it doesn't say anything about the number of arguments, the types of the arguments, or the return type.

```suggestion
  setLastPush: (time: number) => void;
```

Same comment goes for `setExpandList`.

Reply:
Fixed. Thanks!